anndata/core/sparsedataset.py

```diff
@@ -8,79 +8,117 @@
 
 # TODO:
 # - think about supporting the COO format
-from collections.abc import Iterable
+import collections.abc as cabc
 from itertools import accumulate, chain
-from typing import Union, NamedTuple, Tuple, Sequence, Type
+from typing import Union, NamedTuple, Tuple, Sequence, Iterable, Type
```

Comment:
Why this? ABC classes seem like a more useful thing to use for typing.

Reply:
I don’t get this:

![grafik](https://user-images.githubusercontent.com/291575/70048320-4dbc8900-15ca-11ea-87d7-478e10916156.png)

Reply:
The import changed:

```python
# Previously:
from collections.abc import Iterable
# Now:
from typing import Iterable
```

I didn't see why. A similar thing happened in `h5ad.py` with `Mapping`.

Reply:
We’re using `from typing import ...` everywhere. `Iterable` exists in `typing`. `isinstance(x, collections.abc.Iterable)` works, `isinstance(x, typing.Iterable)` doesn’t necessarily.

For consistency we should keep `from typing import ...` and use `import collections.abc as cabc` to access `isinstance`-checkable ABCs.

Reply:
Hmm. My preference would probably be the other way, since the ABCs can be used in more cases. Use `import typing as typ` or something for brevity, but keep the most useful cases the shortest.